hub/core/storage/local.py

```diff
@@ -173,8 +173,9 @@ def _check_is_file(self, path: str):
         Raises:
             DirectoryAtPathException: If a directory is found at the path.
         """
-        full_path = os.path.join(self.root, path)
-        full_path = os.path.expanduser(full_path)
+        full_path = posixpath.join(self.root, path)
+        full_path = posixpath.expanduser(full_path)
+        full_path = str(pathlib.Path(full_path))
```

Comment:
could you elaborate on why this was changed?

Reply:
Otherwise on windows, you will get a mix of windows and posix style paths